pymongo/client_session.py

```diff
@@ -1097,19 +1101,12 @@ def pop_all(self) -> list[_ServerSession]:
             ids.append(self.pop().session_id)
         return ids
 
-    def get_server_session(self, session_timeout_minutes: float) -> _ServerSession:
-        # Although the Driver Sessions Spec says we only clear stale sessions
-        # in return_server_session, PyMongo can't take a lock when returning
-        # sessions from a __del__ method (like in Cursor.__die), so it can't
-        # clear stale sessions there. In case many sessions were returned via
-        # __del__, check for stale sessions here too.
-        self._clear_stale(session_timeout_minutes)
-
+    def get_server_session(self) -> _ServerSession:
         # The most recently used sessions are on the left.
         while self:
-            s = self.popleft()
-            if not s.timed_out(session_timeout_minutes):
-                return s
```

Comment:
This change is necessary because we no longer have access to `session_timeout_minutes` at the time of session creation. Is it feasible to move this timeout check to the moment of session application instead, where we have the underlying connection and access to the timeout?

Reply:
Let's keep this method the same but delay calling it until ClientSession._materialize() is called.